plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -613,19 +601,27 @@ private TestExecutionResult testPluginAgainst(MavenCoordinates coreCoordinates,
         }
     }
 
-    public void cloneFromSCM(PomData pomData, String name, String version, File checkoutDirectory, String tag) throws ComponentLookupException, ScmException, IOException {
-	String scmTag = !(tag.equals("")) ? tag : getScmTag(pomData, name, version);
+    /**
+     *
+     * @deprecated
+     */
+    @Deprecated
+    public void cloneFromSCM(PomData pomData, String name, String version, File checkoutDirectory, String tag)
+            throws IOException {
+        cloneFromSCM(pomData, name, version, checkoutDirectory, tag, Collections.emptyMap());
+    }
+
+    public void cloneFromSCM(PomData pomData, String name, String version, File checkoutDirectory, String tag, Map<String, Object> beforeCheckout) throws IOException {
+	    String scmTag = !("".equals(tag)) ? tag : getScmTag(pomData, name, version);
```

Comment:
```suggestion
	    String scmTag = !tag.isEmpty() ? tag : getScmTag(pomData, name, version);
```

Reply:
why not if we don't care about being NPE free.

Reply:
Is `tag` supposed to be `@CheckForNull`? For that matter, is `""` actually a legitimate argument?

Reply:
I don't know all the history/reasons of this code so I have just avoid changing it. (except avoid a possible NPE).

Reply:
I guess `StringUtils.isNotEmpty(tag))` will do the job